packages/frontend-2/components/auth/sso/Login.vue

```diff
@@ -0,0 +1,187 @@
+import { useDebounceFn } from '@vueuse/core'
+import type { AuthSsoLogin_WorkspaceFragment } from '~/lib/common/generated/gql/graphql'
+import { graphql } from '~/lib/common/generated/gql/gql'
+
+type FormValues = {
+  email: string
+  workspace: AuthSsoLogin_WorkspaceFragment | undefined
+}
+
+enum EmailCheckState {
+  Idle = 'idle',
+  Checking = 'checking',
+  Checked = 'checked'
+}
+
+graphql(`
+  fragment AuthSsoLogin_Workspace on LimitedWorkspace {
+    id
+    slug
+    name
+    logo
+    defaultLogoIndex
+  }
+`)
+
+const { meta, handleSubmit, setFieldValue, values } = useForm<FormValues>({
+  initialValues: {
+    email: '',
+    workspace: undefined
+  }
+})
+
+const { challenge } = useLoginOrRegisterUtils()
+const { signInOrSignUpWithSso } = useAuthManager()
+const logger = useLogger()
+const { triggerNotification } = useGlobalToast()
+
+const loading = ref(false)
+const email = ref('')
+const emailCheckState = ref<EmailCheckState>(EmailCheckState.Idle)
+
+const {
+  loading: isChecking,
+  result,
+  onResult
+} = useQuery(
+  workspaceSsoByEmailQuery,
+  () => ({ email: email.value }),
+  () => ({
+    enabled: emailCheckState.value === 'checking'
+  })
+)
+
+const helpText = computed(() => {
+  if (isChecking.value) return 'Checking SSO availability...'
+  if (availableWorkspaces.value.length === 0 && emailCheckState.value === 'checked') {
+    return 'No SSO-enabled workspaces found for this email'
+  }
+  return undefined
+})
+
+const errorMessage = computed(() => {
+  if (emailCheckState.value === 'checked' && availableWorkspaces.value.length === 0) {
+    return 'This email is not associated with any SSO-enabled workspaces'
+  }
+  return undefined
+})
+
+const availableWorkspaces = computed(() => result.value?.workspaceSsoByEmail || [])
+
+// Show workspace selector only if multiple options exist
+const shouldShowWorkspaceSelector = computed(
+  () =>
+    availableWorkspaces.value.length > 1 &&
+    emailCheckState.value === EmailCheckState.Checked
+)
+
+// Valid when email passes validation and has associated workspaces
+const isValid = computed(
+  () =>
+    meta.value.valid &&
+    emailCheckState.value === 'checked' &&
+    availableWorkspaces.value.length > 0
+)
+
+const buttonText = computed(() => {
+  if (isChecking.value) return 'Checking...'
+  if (!isValid.value) return 'Single Sign-On'
+  return values.workspace?.name ? `Sign in to ${values.workspace.name}` : 'Sign in'
+})
+
+const debouncedCheckEmail = useDebounceFn((value: string) => {
+  if (!value || !meta.value.valid) {
+    emailCheckState.value = EmailCheckState.Idle
+    return
+  }
+  emailCheckState.value = EmailCheckState.Checking
+}, 300)
+
+const onEmailChange = (value: string) => {
+  email.value = value
+  emailCheckState.value = EmailCheckState.Idle
+  setFieldValue('workspace', undefined)
+  debouncedCheckEmail(value)
+}
+
+const onSubmit = handleSubmit((values: FormValues) => {
+  if (!values.workspace) return
```

Comment:
y not use a required rule? 🤔 ideally there shouldnt be any validation in this callback, vee-validate should've already handled it

Reply:
oh and also - y the `values: FormValues`? `values` should already have the correct type from the useForm generic

Reply:
@andrewwallacespeckle well, yea, that's what you've typed the form values as. the field has the isRequired rule on it, so it doesn't appear that it can actually be undefined and you can update the type